05-Mining-Protocol.md

```diff
@@ -1,101 +1,182 @@
-## 5.1 Channels
+## 5.1 Jobs
 
-The protocol is designed such that downstream devices (or proxies) open communication channels with upstream stratum nodes within established connections.
-The upstream stratum endpoints could be actual mining servers or proxies that pass the messages further upstream.
-Each channel identifies a dedicated mining session associated with an authorized user.
-Upstream stratum nodes accept work submissions and specify a mining target on a per-channel basis.
+A **Job** consists of a unit of work containing all the necessary information about the hashing space over some candidate block header.
 
-There can theoretically be up to 2^32 open channels within one physical connection to an upstream stratum node.
-All channels are independent of each other, but share some messages broadcast from the server for higher efficiency (e.g. information about a new prevhash).
-Each channel is identified by its `channel_id` (`U32`), which is consistent throughout the whole life of the connection.
+Each mining device has to work on a unique part of the whole search space.
+The full search space is defined in part by valid values in the following block header fields:
 
-A proxy can either transparently allow its clients to open separate channels with the server (preferred behavior) or aggregate open connections from downstream devices into its own open channel with the server and translate the messages accordingly (present mainly for allowing v1 proxies).
-Both options have some practical use cases.
-In either case, proxies SHOULD aggregate clients' channels into a smaller number of TCP connections.
-This saves network traffic for broadcast messages sent by a server because fewer messages need to be sent in total, which leads to lower latencies as a result.
-And it further increases efficiency by allowing larger packets to be sent.
+- Nonce header field (32 bits)
+- Version header field (16 bits, as specified by [BIP 320](https://github.com/bitcoin/bips/blob/master/bip-0320.mediawiki))
+- Timestamp header field
 
-The protocol defines three types of channels: **standard channels**, **extended channels** (mining sessions) and **group channels** (organizational), which are useful for different purposes.
+The other portion of the block header that is used to define the full search space is the Merkle Root, which is deterministically computed from:
+- Coinbase transaction: typically 4-8 bytes, possibly much more.
```

Comment:
on [his review](https://github.com/plebhash/sv2-spec/commit/e874f121423a2667e8731cdd41a5922697f5578d) @mcelrath pointed out:
> Coinbase tx is much larger than that, do you mean just that extranonce is 4-8 bytes?

This line was originally written by the spec authors. This PR is just moving it to a different section of the doc.

I will leave this question for the spec authors to answer.

Reply:
I don't think coinbase transaction can possibly have as little as 8 bytes. It must be some typo. I'd remove it.